tap_intercom/intercom.py

```diff
@@ -53,23 +79,25 @@ def unixseconds_to_datetime(self, ms: int) -> datetime.datetime:
             datetime.datetime.fromtimestamp(ms, datetime.timezone.utc) if ms else None
         )
 
-    def paginate(self, url: str, params: Dict = None):
+    def paginate(
+        self,
+        url: str,
+        pagination_path: list,
+        tap_stream_id: str,
+        data_field: str,
+        params: Dict = None,
+    ):
         while True:
             data = self.call_api(url, params=params)
             if data:
-                yield from data.get(self.DATA_FIELD.get(self.tap_stream_id))
-            if self.tap_stream_id == "contacts":
-                starting_after = self.get_value(
-                    data, ["pages", "next", "starting_after"]
-                )
-                if not starting_after:
-                    return
-                params["starting_after"] = starting_after
-
-            else:
-                url = self.get_value(data, ["pages", "next"])
-            if not url or url == "null":
+                yield from data.get(data_field)
+            next_page = self.get_value(data, pagination_path)
+            if not next_page or next_page == "null":
                 return
+            if tap_stream_id == "contacts":
+                params["starting_after"] = next_page
+            else:
+                url = next_page
```

Comment:
the `params = {}` line you added after my previous comment is actually still prudent